go/vt/vttablet/heartbeat/reader.go

```diff
@@ -80,7 +80,6 @@ func (r *Reader) Open(dbc dbconfigs.DBConfigs) {
 	r.mu.Lock()
 	defer r.mu.Unlock()
 	if r.isOpen {
-		log.Fatalf("BUG: Reader object cannot be initialized twice without closing in between: %v", r)
```

Comment:
Can the original problem be generalized as follows?

> Open() can be called repeatedly by Tabletserver.serveNewType() athough the tablet type did not change (but the query service state changed).

IMO this is a missing functionality in `serveNewType()` which we should add eventually. That function should only open or close the heartbeat reader and writer when it sees transitions from the MASTER to non-MASTER tablet type and vice versa. With that in place, repeated Open()s should no longer be happening.

For now I am fine with just documenting why we allow repeated calls to Open().